lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1892,21 +1893,22 @@ def increment_failed_requests_count(code)
     # ingested by the Stackdriver Logging API.
     def increment_ingested_entries_count(count)
       return unless @ingested_entries_count
-      @ingested_entries_count.increment({}, count)
+      @ingested_entries_count.increment({ grpc: @use_grpc, code: @ok_code },
+                                        count)
     end
 
     # Increment the metric for the number of log entries that were dropped
     # and not ingested by the Stackdriver Logging API.
-    def increment_dropped_entries_count(count)
+    def increment_dropped_entries_count(count, code)
       return unless @dropped_entries_count
-      @dropped_entries_count.increment({}, count)
+      @dropped_entries_count.increment({ grpc: @use_grpc, code: code }, count)
```

Comment:
Rather than sending a boolean to help us interpret what the code means, should we just add two labels, `http_status` and `grpc_error_code`, and only set one of them depending on the value of `@use_grpc`?

Reply:
Does that imply we will have two separate Monarch metrics or two sets of labels in Monarch as well? It feels more difficult to maintain and filter that way IMO.

Reply:
It means we'd have a single metric with two metric labels, only one of which would be set at a time... For gRPC, we'd submit `dropped_entries_count{grpc_error_code=...}`, and for REST it would have `dropped_entries_count{http_status=...}`.

Reply:
Oh, never mind. I just realized that we've always had the `grpc` metric label. So all we're doing here is adding error code tracking. Please ignore this thread.

Reply:
Cool cool.

Reply:
Actually, looks like I got confused about my earlier confusion. You're actually adding the `grpc` label here. So this thread still applies.

Reply:
[Previously](Reference: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb#L1917) we only have `use_grpc` label for `increment_successful_requests_count` and `increment_failed_requests_count`. Here the same label are added for the other three metrics.

At least for `increment_ingested_entries_count`, we have to provide a way to infer the right code (`0` for `gRPC` and `200` for `REST`). The options are:
1. Explicitly pass `0` or `200` as the `OK` code. This way we don't care about the `use_grpc` boolean.
2. Skip the `0` or `200` code but pass in a `use_grpc` boolean, so later on we can infer the code from the boolean.
3. Pass in both the boolean and the code explicitly. This way minimum logic is required when translating the metrics. But it might be a bit redundant.

Because of the fact that we already have `use_grpc` for 2 out of the 5 metrics, the option 2 feels more natural to me.

Reply:
Hmm, why would we care about the _error_ code for _successfully_ ingested metrics?

Reply:
Ok, never mind, I see. We're already combining the metrics, so the only way to find out which ones are successful would be to look at the error code. Sorry for my earlier confusion.
The most consistent thing to do in this case is actually option 3 -- that way all metrics have the same labels. Again, sorry to make you redo the work, but you can reinstate https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/186/commits/3667b07eb17ce5254e3643db914b05b75d1f13d2.

Reply:
Right. We are combining the metrics in the next step (Collectd translation).

No worries. I've just restored that commit (I kept it in its own commit because I was debating between option 2 and option 3).